app/helpers/katello/subscription_mailer_helper.rb

```diff
@@ -0,0 +1,18 @@
+module Katello
+  module SubscriptionMailerHelper
+    include Rails.application.routes.url_helpers
+
+    def report_url
+      base_url = report_data_report_template_path(@report_template.id, job_id: @provider_job_id) unless @pools.empty?
```

Comment:
It would be better to not have this code know anything about @pools since it doesn't use that value if it's not empty

a clearer check would be `if @report_template && @provider_job_id`